base/src/H264EncoderV4L2Helper.cpp

```diff
@@ -200,10 +236,100 @@ int H264EncoderV4L2Helper::setExtControls(v4l2_ext_control &control)
     return v4l2_ioctl(mFD, VIDIOC_S_EXT_CTRLS, &ctrls);
 }
 
+int
+H264EncoderV4L2Helper::getExtControls(v4l2_ext_controls &ctl)
+{
+    int ret;
+
+    ret = v4l2_ioctl(mFD, VIDIOC_G_EXT_CTRLS, &ctl);
+
+    return ret;
+}
+
+int
+H264EncoderV4L2Helper::getMotionVectors(uint32_t buffer_index,
+        v4l2_ctrl_videoenc_outputbuf_metadata_MV &enc_mv_metadata)
+{
+    v4l2_ctrl_video_metadata metadata;
+    struct v4l2_ext_control control;
+    struct v4l2_ext_controls ctrls;
+
+    ctrls.count = 1;
+    ctrls.controls = &control;
+    ctrls.ctrl_class = V4L2_CTRL_CLASS_MPEG;
+
+    metadata.buffer_index = buffer_index;
+    metadata.VideoEncMetadataMV = &enc_mv_metadata;
+
+    control.id = V4L2_CID_MPEG_VIDEOENC_METADATA_MV;
+    control.string = (char *)&metadata;
+
+    getExtControls(ctrls);
+}
+
+void H264EncoderV4L2Helper::serializeMotionVectors(v4l2_ctrl_videoenc_outputbuf_metadata_MV enc_mv_metadata, frame_container &frames)
+{
+    uint32_t numMVs = enc_mv_metadata.bufSize / sizeof(MVInfo);
+    MVInfo *pInfo = enc_mv_metadata.pMVInfo;
+
+    std::vector<CircleOverlay> circleOverlays;
+    CompositeOverlay compositeOverlay;
+
+    int totalMacroblockInRow = floor(mWidth / 16);
+    auto motionVectorFrame = makeFrame(1024 * 1024 * 3, motionVectorFramePinId);
+    uint32_t *frameBuffer = reinterpret_cast<uint32_t *>(motionVectorFrame->data());
+    size_t mCount = 0;
+    for (uint32_t i = 0; i < numMVs; i++, pInfo++)
+    {
+
+        if (abs(pInfo->mv_x) > motionVectorThreshold || abs(pInfo->mv_y) > motionVectorThreshold)
+        {
+            auto tempY = floor(i / totalMacroblockInRow);
+            auto y = tempY * 16 + 8;
```

Comment:
16 and 8 are harcoded values - why ? at least, put comments explaining these

Reply:
Added comments explaining about the values